src/clr-angular/popover/signpost/signpost.spec.ts

```diff
@@ -57,6 +57,95 @@ export default function(): void {
       });
     });
 
+    describe('focus management', function() {
+      spec(ClrSignpost, TestDefaultSignpost, ClrSignpostModule);
+
+      beforeEach(function(this: Context) {
+        this.ifOpenService = this.getClarityProvider(IfOpenService);
+      });
+
+      it('should not get focus on trigger initially', function(this: Context) {
+        const signpostToggle: HTMLElement = this.hostElement.querySelector('.signpost-action');
+        this.ifOpenService.open = false;
+        this.detectChanges();
+        expect(signpostToggle).not.toBeNull();
+        expect(document.activeElement).not.toBe(signpostToggle);
+      });
+
+      it('should focus signpost content body on opening', function(this: Context) {
+        this.ifOpenService.open = true;
+        this.detectChanges();
+        const signpostContentBody: HTMLElement = this.hostElement.querySelector('.signpost-content-body');
+        expect(signpostContentBody).not.toBeNull();
+        expect(document.activeElement).toBe(signpostContentBody);
+      });
+
+      it('should not get focus back on trigger if signpost gets closed with outside click on another interactive element', function(this: Context) {
+        this.ifOpenService.open = true;
+        this.detectChanges();
+        let signpostContent: HTMLElement = this.hostElement.querySelector('.signpost-content');
+        expect(signpostContent).not.toBeNull();
+
+        // dynamic click doesn't set the focus so here manually focusing first
+        this.hostComponent.outsideClickBtn.nativeElement.focus();
+        this.hostComponent.outsideClickBtn.nativeElement.click();
+        this.detectChanges();
+
+        signpostContent = this.hostElement.querySelector('.signpost-content');
+        expect(signpostContent).toBeNull();
```

Comment:
nit: don't see why we need a variable here when its only used once in both cases. This appears to be true in several tests, skip the local variable so it isn't confused.

Reply:
it's because the reference doesn't get updated when the dom gets changes. We need to retrieve it again to get the updated result for the selector.

Reply:
I know, my suggestion was to just query the selector twice in the expect instead of putting it in a variable first. That was confusing to me at first.

Reply:
Done